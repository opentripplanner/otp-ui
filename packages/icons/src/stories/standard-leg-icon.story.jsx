import React from "react";

import StandardLegIcon from "../standard-leg-icon";

import LegIconRenderer from "./leg-icon-renderer";

export default {
  title: "Icons/StandardLegIcon",
  component: StandardLegIcon
};

export const StandardLegIconExamples = () => {
  return <LegIconRenderer component={StandardLegIcon} />;
};
