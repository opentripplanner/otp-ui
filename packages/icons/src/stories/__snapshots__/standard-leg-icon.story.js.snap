// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icons/StandardLegIcon StandardLegIconExamples smoke-test 1`] = `
<table class="icon-renderer__StyledTable-sc-epf3ui-0 fqtAeK">
  <thead>
    <tr>
      <th>
        Leg Type
      </th>
      <th>
        Result
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        Personal bike
      </td>
      <td>
        <div>
          <svg version="1.1"
               viewbox="0 0 32 32"
               height="100%"
               width="100%"
               leg="[object Object]"
          >
            <g>
              <path d="M6.9,15.4c-3.8,0-6.9,3.1-6.9,6.9c0,3.8,3.1,6.9,6.9,7c3.8,0,6.9-3.1,6.9-7C13.9,18.6,10.8,15.5,6.9,15.4z M10.7,26.1 c-0.9,1-2.3,1.6-3.7,1.6c-1.4,0-2.8-0.6-3.7-1.6c-0.9-0.9-1.5-2.3-1.5-3.7c0-1.4,0.6-2.8,1.5-3.7c1-1,2.3-1.5,3.7-1.5 c1.4,0,2.8,0.6,3.7,1.5c1,0.9,1.5,2.3,1.5,3.7C12.2,23.8,11.6,25.2,10.7,26.1z">
              </path>
              <path d="M19.2,7.7c1.4,0,2.5-1.1,2.5-2.5c0-1.4-1.1-2.5-2.5-2.5c-1.4,0-2.5,1.1-2.5,2.5C16.7,6.6,17.8,7.7,19.2,7.7z">
              </path>
              <path d="M23.9,13.9c1.5,0,1.5-2-0.1-2h-4l-2.8-4.6c-0.6-0.8-2.3-0.9-3-0.1l-4.9,4.9C8,13.3,8.7,15,10,15.4l4.5,2.8v4.6 c0,1,0.8,1.6,1.5,1.6c0.7,0,1.5-0.5,1.5-1.6V17c0-0.5-0.1-1.2-0.5-1.3l-3-1.9l2.5-2.6l1.5,2c0.3,0.6,0.9,0.8,1.5,0.8H23.9z">
              </path>
              <g>
                <path d="M25.1,15.3c-3.8,0-6.9,3.1-6.9,6.9c0,3.8,3.1,6.9,6.9,6.9c3.8,0,6.9-3.1,6.9-6.9C32,18.5,28.9,15.3,25.1,15.3z M28.8,26 c-0.9,0.9-2.3,1.5-3.7,1.5c-1.4,0-2.8-0.6-3.7-1.5c-0.9-1-1.5-2.3-1.5-3.7c0-1.5,0.6-2.8,1.5-3.7c1-1,2.3-1.5,3.7-1.5 c1.5,0,2.8,0.6,3.7,1.5c1,1,1.5,2.3,1.5,3.7C30.3,23.7,29.7,25,28.8,26z">
                </path>
              </g>
            </g>
          </svg>
        </div>
      </td>
    </tr>
    <tr>
      <td>
        Bike rental
      </td>
      <td>
        <div>
          <svg version="1.1"
               viewbox="0 0 32 32"
               height="100%"
               width="100%"
               leg="[object Object]"
          >
            <g>
              <path d="M6.9,15.4c-3.8,0-6.9,3.1-6.9,6.9c0,3.8,3.1,6.9,6.9,7c3.8,0,6.9-3.1,6.9-7C13.9,18.6,10.8,15.5,6.9,15.4z M10.7,26.1 c-0.9,1-2.3,1.6-3.7,1.6c-1.4,0-2.8-0.6-3.7-1.6c-0.9-0.9-1.5-2.3-1.5-3.7c0-1.4,0.6-2.8,1.5-3.7c1-1,2.3-1.5,3.7-1.5 c1.4,0,2.8,0.6,3.7,1.5c1,0.9,1.5,2.3,1.5,3.7C12.2,23.8,11.6,25.2,10.7,26.1z">
              </path>
              <path d="M19.2,7.7c1.4,0,2.5-1.1,2.5-2.5c0-1.4-1.1-2.5-2.5-2.5c-1.4,0-2.5,1.1-2.5,2.5C16.7,6.6,17.8,7.7,19.2,7.7z">
              </path>
              <path d="M23.9,13.9c1.5,0,1.5-2-0.1-2h-4l-2.8-4.6c-0.6-0.8-2.3-0.9-3-0.1l-4.9,4.9C8,13.3,8.7,15,10,15.4l4.5,2.8v4.6 c0,1,0.8,1.6,1.5,1.6c0.7,0,1.5-0.5,1.5-1.6V17c0-0.5-0.1-1.2-0.5-1.3l-3-1.9l2.5-2.6l1.5,2c0.3,0.6,0.9,0.8,1.5,0.8H23.9z">
              </path>
              <g>
                <path d="M25.1,15.3c-3.8,0-6.9,3.1-6.9,6.9c0,3.8,3.1,6.9,6.9,6.9c3.8,0,6.9-3.1,6.9-6.9C32,18.5,28.9,15.3,25.1,15.3z M28.8,26 c-0.9,0.9-2.3,1.5-3.7,1.5c-1.4,0-2.8-0.6-3.7-1.5c-0.9-1-1.5-2.3-1.5-3.7c0-1.5,0.6-2.8,1.5-3.7c1-1,2.3-1.5,3.7-1.5 c1.5,0,2.8,0.6,3.7,1.5c1,1,1.5,2.3,1.5,3.7C30.3,23.7,29.7,25,28.8,26z">
                </path>
              </g>
            </g>
          </svg>
        </div>
      </td>
    </tr>
    <tr>
      <td>
        Bus
      </td>
      <td>
        <div>
          <svg version="1.1"
               viewbox="0 0 32 32"
               height="100%"
               width="100%"
               leg="[object Object]"
          >
            <g>
              <g>
                <g>
                  <path d="M19.9,14.4c-1.1,0-2,0.7-2.3,1.7c0.3,0,4.3,0,4.6,0C21.9,15.1,21,14.4,19.9,14.4z">
                  </path>
                  <path d="M26.9,7c-0.2-0.7-0.5-1.4-1.2-1.8c-5.2-2.7-14.4-2.6-19.1,0C6.2,5.5,5.4,6.1,5.1,7c-0.4,1.5-0.6,3-0.6,3.7l0,5.7l0,2.2 v6.1c0,0.2,0.2,0.3,0.4,0.3h22.3c0.2,0,0.3-0.1,0.3-0.3v-5.2l0-8.9C27.5,9.7,27.3,8.1,26.9,7z M11,4.9h9.9V6H11V4.9z M7.9,21.4 c-1,0-1.8-0.8-1.8-1.8c0-0.5,0.2-0.9,0.5-1.3C7,18,7.5,17.8,7.9,17.8c1,0,1.8,0.8,1.8,1.8C9.7,20.6,8.9,21.4,7.9,21.4z M24,21.4 c-1,0-1.8-0.8-1.8-1.8c0-1,0.8-1.8,1.8-1.8c1,0,1.8,0.8,1.8,1.8C25.8,20.6,25,21.4,24,21.4z M25.2,16.4H6.5C6.2,16.4,6,16,6,15.8 l0.5-7.9c0-0.4,0.3-0.8,0.8-0.8h17.4c0.2,0,0.7,0.5,0.7,1l0.6,7.7C26,16.1,25.6,16.4,25.2,16.4z">
                  </path>
                </g>
                <g>
                  <path d="M6.5,27.6c0,0.8,0.8,1.2,1.6,1.2c0.4,0,0.8-0.1,1-0.3c0.3-0.2,0.4-0.5,0.4-0.9c0,0,0-1.7,0-1.9c-0.2,0-2.8,0-3,0 C6.5,25.9,6.5,27.6,6.5,27.6z">
                  </path>
                  <path d="M22.3,27.6c0,0.8,0.8,1.1,1.6,1.1c0.8,0,1.6-0.4,1.6-1.1c0,0,0-1.7,0-1.8c-0.2,0-3.1,0-3.2,0 C22.3,25.9,22.3,27.6,22.3,27.6z">
                  </path>
                </g>
              </g>
            </g>
          </svg>
        </div>
      </td>
    </tr>
    <tr>
      <td>
        Micromobility (from Razor company)
      </td>
      <td>
        <div>
          <svg height="100%"
               viewbox="0 0 512 512"
               width="100%"
               xmlns="http://www.w3.org/2000/svg"
               leg="[object Object]"
          >
            <path d="m464.027344 393.210938-47.425782-341.464844c-4.097656-29.5-29.648437-51.746094-59.429687-51.746094h-68.171875v40h68.171875c9.929687 0 18.441406 7.414062 19.808594 17.246094l36.148437 260.25c-52.128906 15.089844-91.835937 59.722656-99.703125 114.503906h-196.867187c-8.253906-23.277344-30.484375-40-56.558594-40-33.085938 0-60 26.914062-60 60s26.914062 60 60 60c26.074219 0 48.304688-16.722656 56.558594-40h235.441406v-20c0-43.472656 27.886719-80.550781 66.710938-94.296875l5.699218 41.042969c-19.234375 10.003906-32.410156 30.113281-32.410156 53.253906 0 33.085938 26.914062 60 60 60s60-26.914062 60-60c0-28.964844-20.632812-53.203125-47.972656-58.789062zm-404.027344 78.789062c-11.027344 0-20-8.972656-20-20s8.972656-20 20-20 20 8.972656 20 20-8.972656 20-20 20zm392 0c-11.027344 0-20-8.972656-20-20s8.972656-20 20-20 20 8.972656 20 20-8.972656 20-20 20zm0 0">
            </path>
          </svg>
        </div>
      </td>
    </tr>
    <tr>
      <td>
        Park &amp; Ride
      </td>
      <td>
        <div>
          <svg xmlns="http://www.w3.org/2000/svg"
               viewbox="0 0 512 512"
               height="100%"
               width="100%"
               leg="[object Object]"
          >
            <path d="M499.991 168h-54.815l-7.854-20.944c-9.192-24.513-25.425-45.351-46.942-60.263S343.651 64 317.472 64H194.528c-26.18 0-51.391 7.882-72.908 22.793-21.518 14.912-37.75 35.75-46.942 60.263L66.824 168H12.009c-8.191 0-13.974 8.024-11.384 15.795l8 24A12 12 0 0 0 20.009 216h28.815l-.052.14C29.222 227.093 16 247.997 16 272v48c0 16.225 6.049 31.029 16 42.309V424c0 13.255 10.745 24 24 24h48c13.255 0 24-10.745 24-24v-40h256v40c0 13.255 10.745 24 24 24h48c13.255 0 24-10.745 24-24v-61.691c9.951-11.281 16-26.085 16-42.309v-48c0-24.003-13.222-44.907-32.772-55.86l-.052-.14h28.815a12 12 0 0 0 11.384-8.205l8-24c2.59-7.771-3.193-15.795-11.384-15.795zm-365.388 1.528C143.918 144.689 168 128 194.528 128h122.944c26.528 0 50.61 16.689 59.925 41.528L391.824 208H120.176l14.427-38.472zM88 328c-17.673 0-32-14.327-32-32 0-17.673 14.327-32 32-32s48 30.327 48 48-30.327 16-48 16zm336 0c-17.673 0-48 1.673-48-16 0-17.673 30.327-48 48-48s32 14.327 32 32c0 17.673-14.327 32-32 32z">
            </path>
          </svg>
        </div>
      </td>
    </tr>
    <tr>
      <td>
        TNC (Uber)
      </td>
      <td>
        <div>
          <svg xmlns="http://www.w3.org/2000/svg"
               viewbox="0 0 512 512"
               height="100%"
               width="100%"
               leg="[object Object]"
          >
            <path d="M499.991 168h-54.815l-7.854-20.944c-9.192-24.513-25.425-45.351-46.942-60.263S343.651 64 317.472 64H194.528c-26.18 0-51.391 7.882-72.908 22.793-21.518 14.912-37.75 35.75-46.942 60.263L66.824 168H12.009c-8.191 0-13.974 8.024-11.384 15.795l8 24A12 12 0 0 0 20.009 216h28.815l-.052.14C29.222 227.093 16 247.997 16 272v48c0 16.225 6.049 31.029 16 42.309V424c0 13.255 10.745 24 24 24h48c13.255 0 24-10.745 24-24v-40h256v40c0 13.255 10.745 24 24 24h48c13.255 0 24-10.745 24-24v-61.691c9.951-11.281 16-26.085 16-42.309v-48c0-24.003-13.222-44.907-32.772-55.86l-.052-.14h28.815a12 12 0 0 0 11.384-8.205l8-24c2.59-7.771-3.193-15.795-11.384-15.795zm-365.388 1.528C143.918 144.689 168 128 194.528 128h122.944c26.528 0 50.61 16.689 59.925 41.528L391.824 208H120.176l14.427-38.472zM88 328c-17.673 0-32-14.327-32-32 0-17.673 14.327-32 32-32s48 30.327 48 48-30.327 16-48 16zm336 0c-17.673 0-48 1.673-48-16 0-17.673 30.327-48 48-48s32 14.327 32 32c0 17.673-14.327 32-32 32z">
            </path>
          </svg>
        </div>
      </td>
    </tr>
    <tr>
      <td>
        Tram
      </td>
      <td>
        <div>
          <svg version="1.1"
               viewbox="0 0 32 32"
               height="100%"
               width="100%"
               leg="[object Object]"
          >
            <path d="M9.9,27.2v3.2L8.5,32h1.8l0.8-1.1v-1.8H21v1.8l1,1.1h1.6l-1.4-1.7v-3.1h3.4l0.3-1.6V5.2h-3.2l-0.7-1.9h-5.3V2.6l2.7-1.7V0 h-6.5v1l2.6,1.6v0.7h-5.3L9.4,5.2H6.2l0,20.4l0.3,1.6H9.9z M22.9,8.1l1.8,1.1V16l-1.8,0.6V8.1z M14.7,0.9h2.7L16,1.8L14.7,0.9z M9.9,8.3c0-0.5,0.3-0.8,0.8-0.8h10.4c0.5,0,0.8,0.3,0.8,0.8v7.8c0,0.5-0.3,0.8-0.8,0.8H10.7c-0.5,0-0.8-0.3-0.8-0.8V8.3z M21.9,18.1V20h-3.2v-1.9H21.9z M13.2,18.1v1.8H10v-1.8H13.2z M22.8,21.1v1.6H8.9v-1.6H22.8z M7.2,9.1L9,8v8.6L7.2,16V9.1z">
            </path>
          </svg>
        </div>
      </td>
    </tr>
    <tr>
      <td>
        Walk
      </td>
      <td>
        <div>
          <svg version="1.1"
               viewbox="0 0 32 32"
               height="100%"
               width="100%"
               leg="[object Object]"
          >
            <g>
              <path d="M17.1,0c1.5,0,2.7,1.2,2.7,2.7c0,1.5-1.2,2.7-2.7,2.7s-2.7-1.2-2.7-2.7C14.5,1.2,15.7,0,17.1,0z M25.2,16.6 c-0.4,0.5-1.1,0.7-1.7,0.3h0c0,0-3.7-2.5-3.9-2.7c-0.3-0.2-0.4-0.6-0.5-0.8c-0.1-0.2-0.5-1-0.8-1.5L18,13.3l-0.9,3.9 c0,0,4.6,5.4,4.7,5.5c0.1,0.1,0.2,0.5,0.3,0.7c0.1,0.2,1.2,6.6,1.2,6.6c0.2,1-0.5,1.9-1.4,2.1c-1,0.2-1.9-0.5-2.1-1.4l-1.2-6.3 l-3.8-4.1c0,0-0.8,3.9-0.9,4c0,0.1-0.1,0.2-0.1,0.3c0,0.1-3.9,6.7-3.9,6.7c-0.5,0.9-1.6,1.1-2.5,0.6c-0.9-0.5-1.1-1.6-0.6-2.5 l3.4-5.8L13,11l-1.9,1.5l-1,4.5c0,0,0,0,0,0c0,0.6-0.7,1.1-1.4,0.9c-0.6-0.1-1-0.4-0.9-1.4h0c0,0,1.1-5,1.1-5.2 c0.1-0.2,0.2-0.5,0.3-0.6c0.8-0.6,3.8-3.1,4.6-3.8c1-0.8,1.5-1.1,2.3-1.1c0.7,0,1.4,0.2,2.1,1C18.7,7.1,19,7.7,19.1,8 c0.2,0.3,2.2,4.4,2.2,4.4l3.6,2.5c0,0,0,0,0,0C25.5,15.3,25.6,16,25.2,16.6z">
              </path>
            </g>
          </svg>
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;
